src/psij/testing/service.py

```diff
@@ -442,9 +441,17 @@ def getSite(self, site_id):
             for oneTest in testContainer['tests']:
 
                 testName = oneTest['test_name']
-                resBySiteIdAndTestName[testName] = oneTest['results']
-                resBySiteIdAndTestName[testName]['branch'] = oneTest['branch']
-                resBySiteIdAndTestName[testName]['test_start_time'] = oneTest['test_start_time']
+
+                foundMatch = 0
+
+                for test in tests_to_match:
+                    if testName == test:
+                        foundMatch = 1
```

Comment:
That could also be simplified to:

```Python
foundMatch = test in tests_to_match
```

That said, you may want to do the filtering through the database query:

```Python
tests = Test.objects(site_id=site_id, run_id=run_id,
                                 branch=run.branch,
                                 test_name__in=tests_to_match).order_by('-test_start_time')
```

Then your result would automatically only contain the desired test names and it will prevent moving the data from the database to the service process for tests that don't match.

Reply:
Thanks for the filtering query.  I've tested it and it works!